examples/platform_channel/macos/Runner/MainFlutterWindow.swift

```diff
@@ -0,0 +1,84 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Cocoa
+import FlutterMacOS
+
+class MainFlutterWindow: NSWindow, FlutterStreamHandler, PowerSourceStateChangeDelegate {
+  private let powerSource = PowerSource()
+  private let stateChangeSource = PowerSourceStateChangeHandler()
+  private var eventSink: FlutterEventSink?
+
+  override func awakeFromNib() {
+    let flutterViewController = FlutterViewController.init()
+    let windowFrame = self.frame
+    self.contentViewController = flutterViewController
+    self.setFrame(windowFrame, display: true)
+
+    // Register battery method channel.
+    let registrar = flutterViewController.registrar(forPlugin: "BatteryLevel")
+    let batteryChannel = FlutterMethodChannel(
+      name: "samples.flutter.io/battery",
+      binaryMessenger: registrar.messenger)
+    batteryChannel.setMethodCallHandler({ [weak self] (call, result) in
+      switch call.method {
+      case "getBatteryLevel":
+        let level = self?.powerSource.getCurrentCapacity()
+        if level == -1 {
+          result(
+            FlutterError(
+              code: "UNAVAILABLE",
+              message: "Battery info unavailable",
+              details: nil))
+        }
+        result(level)
+      default:
+        result(FlutterMethodNotImplemented)
+      }
+    })
+
+    // Register charging event channel.
+    let chargingChannel = FlutterEventChannel(
+      name: "samples.flutter.io/charging",
+      binaryMessenger: registrar.messenger)
+    chargingChannel.setStreamHandler(self)
```

Comment:
will `chargingChannel` remove `self` from handlers list when it's gone?

Reply:
Closing the window exits the application and the channel persists until process exit; we never deregister them.